Charles/caption float

chaznied opened the pull request:
## JIRA Ticket
[https://tbnextjs.atlassian.net/browse/NJP-2114]

## Describe Changes in this Pull Request

- Updated caption styles in fullscreen for video/images/daf images

## Testing Instructions

- Pull this from the tollbrothers-ui repo [https://github.com/tollbros/tollbrothers-ui/tree/charles/captionFloat]
- Run `npm run start`
- Pull this from the tollbrothers.com repo [https://bitbucket.org/tollbrothers/com.tollbrothers.www/branch/charles/NJP-2114-tollbrothers-ui-advantage-page-]
- Update `package.json` with `"@tollbrothers/tollbrothers-ui": "file:../tollbrothers-ui",`
- Run `npm i`
- Run `npm run dev`
- Test here - [http://localhost:3000/advantage]
- Test Daf images here - [http://localhost:3000/luxury-homes-for-sale/California/Skylar-by-Toll-Brothers/Quick-Move-In/261420]
- Test randomly here - [http://localhost:3000/luxury-homes-for-sale/New-Jersey/Canter-Square/Mueller]

## Screen Shot(s)/Video(s) of Working Changes
Please add screen shots or screen recordings showing the changes

### Tasks:

1. [ ] Changes Meet JIRA Tickets Requirements
2. [ ] Commit message with `fix` or `feat` to trigger new release
3. [ ] Tested changes on Mobile, Tablet and Desktop screen sizes
4. [ ] Tested changes on Chrome, FireFox and Safari
5. [ ] Demoed to Design Team (For new features only)
6. [ ] Notified Digital Marketing Team if changes will impact A/B Tests
7. [ ] Checked Tracking/Analytics (Notified Digital Marketing Team if changes impact tracking)

Commit:
fix: started fixing floating caption in fullscreen gallery comp

Commit:
fix: NJP-2114 - fixing captions in full screen mode for video/images/daf images

cormdog commented:
Needs a little tweaking on mobile. Screenshot attached.
![IMG_0564](https://github.com/tollbros/tollbrothers-ui/assets/2752502/49d137a2-614d-4a8d-a557-5bd37b6357e3)

mikeduarte commented on src/components/FullScreenGallery.module.scss:
```diff
@@ -46,7 +46,8 @@
         height: 100%;
         margin: auto;
-        object-fit: contain;
+        object-fit: cover;
```
This causes a really big change to the tb.com gallery, especially on mobile. We might want to get input from Justin.

Before:
![Screenshot 2024-05-15 at 4 32 16 PM](https://github.com/tollbros/tollbrothers-ui/assets/15040013/0562d564-6d98-403e-82a6-b8558aee1c6d)


Your Branch:
![Screenshot 2024-05-15 at 4 32 09 PM](https://github.com/tollbros/tollbrothers-ui/assets/15040013/9e115b9a-6f41-4eec-987f-02563911a1a5)

mikeduarte commented on src/components/GalleryMedia.js:
```diff
@@ -15,7 +15,9 @@ function GalleryMedia({
   onLoad,
-  backgroundColor
+  backgroundColor,
+  isDesignerAppointed
```
I don't think this is being use.

mikeduarte commented on src/components/FullScreenGallery.js:
```diff
@@ -29,7 +29,8 @@ export const FullScreenGallery = ({
   backgroundColor,
-  portalId
+  portalId,
+  isDesignerAppointed
```
Similar to the other PR, we might want to pass a classes prop from the parent app to override styles here. We can huddle about it.

mikeduarte reviewed (changes requested):
Left a few comments

Commit:
fix: NJP-2114 - adjusting video caption for video/walkthrough and adding class as prop to styling from parent

chaznied commented:
@mikeduarte  - updated, only adjusting captions for videos/walkthrough above 768 and added class prop

Chenzo commented:
Hey, per our conversation, fix this from floating over the controls (via Justin's fix of removing all that from any video)

![Screenshot 2024-05-28 at 11 35 43 AM](https://github.com/tollbros/tollbrothers-ui/assets/505404/bf888192-b942-4bcb-94ec-2830dd510dae)

Chenzo reviewed (changes requested):
see my previous comment

chaznied commented:
@Chenzo , @mikeduarte  - I just remembered we can't remove captions from any video because of the disclaimer i had to add for all inside maps...

Commit:
fix: updating with Justins and Vinces input, removing captions from fullscreen videos, removing counter and socials from walkthroughs and videos

chaznied commented:
@Chenzo , @mikeduarte  - Updated GalleryMedia.js to only show captions for images and insidemaps walkthroughs and removing counter and social from videos and walkthroughs

mikeduarte commented on src/components/GalleryMedia.js:
```diff
@@ -25,20 +26,25 @@ function GalleryMedia({
   if (media?.type?.includes('video')) {
     modelLink = null
     type = 'video'
+    customClass = styles.mediaVideo
```
Do we need this if we aren't showing captions on videos?

mikeduarte commented on src/components/FullScreenGallery.module.scss:
```diff
@@ -47,6 +47,13 @@
+
+      iframe {
+        @include breakpoint(sm) {
+          object-fit: cover;
```
I don't think we need this. I don't see a difference when I comment it out.

mikeduarte commented on src/components/GalleryMedia.js:
```diff
@@ -25,20 +26,25 @@ function GalleryMedia({
     modelLink = null
     type = 'walkthrough'
     iframeSrc = getWalkthroughURL(media)
+    customClass = styles.mediaVideo
```
Maybe this should only get applied if there is a caption and showCaption is true?

Commit:
code clean up

Commit:
fix: NJP-2114 Tweak breakpoint, fix control blocking

chaznied commented:
@Chenzo  - I kept the caption for inside maps (legal) and walkthroughs (Justin) and images, removed captions from videos (Justin), removed social/counter from video and walkthroughs (Justin). Images remain the same

Chenzo commented:
Matterports don't seem to work. We chatted. Waiting on fix.

Chenzo reviewed (approved):
It's totally something from Matterport in either your cookies, session or local storage.

sarameix commented:
:tada: This PR is included in version 1.21.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@tollbrothers/tollbrothers-ui/v/1.21.1)
- [GitHub release](https://github.com/tollbros/tollbrothers-ui/releases/tag/v1.21.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
